### PR TITLE
DIG-1198: GitHub action: eslint and prettier for candig-data-portal

### PR DESCRIPTION
**Pull request by OrdiNeu**
## Ticket(s)
[DIG-1198](https://candig.atlassian.net/browse/DIG-1198)

## Description
This adds a ESLint/Prettier check to any PR to develop (and hopefully triggers for this PR)

## Expected Behaviour
An ESLint/Prettier check should show up in all upcoming PRs

## Types of Change(s)
- [x] ✨ New feature (non-breaking change that adds functionality)

[DIG-1198]: https://candig.atlassian.net/browse/DIG-1198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by OrdiNeu** [comment]
Three notes:
1. The linter requires exhaustive dependency arrays in its useEffect hooks, however, we often don't want to re-run the effect when some of the variables change.
2. The linter really dislikes my promise chaining in SearchHandler.js, and assumes I want to keep it between renders, which I don't.
3. The linter is also firing off on some unused functions, like mcode.js. Unsure if we wanted to remove those files or keep them as reference, so I just disabled es-lint on those lines.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Lint
+
+on:
+    # Trigger the workflow on push or pull request,
+    # but only for the main branch
```
Comment (CourtneyGosselin) [inline .github/workflows/lint.yml]: Comment should be for develop branch

File: src/views/clinical/mcode.js
Comment (CourtneyGosselin) [inline src/views/clinical/mcode.js:1]: I am pretty sure this file can just be entirely deleted

File: src/views/dashboard/Default/datasetIdSelect.js
Comment (CourtneyGosselin) [inline src/views/dashboard/Default/datasetIdSelect.js:1]: Do we still need this file?

File: src/views/genomicsData/VariantsSearch.js
Comment (CourtneyGosselin) [inline src/views/genomicsData/VariantsSearch.js:1]: Do we still want to keep this file if this varientSearch is not used?

**Review by CourtneyGosselin — CHANGES_REQUESTED** [review]
Overall LGTM just has some files that may need to be removed and one comment mentioning the wrong branch

**Review by CourtneyGosselin — APPROVED** [review]
LGTM

**Pull request merged**